src/Button/Button.react.js

```diff
@@ -56,15 +56,15 @@ const contextTypes = {
 };
 
 function getStyles(props, context, state) {
-    const { button, buttonFlat, buttonRaised, buttonDisabled, buttonRaisedDisabled } = context.uiTheme;
+    const { button, buttonFlat, buttonRaised } = context.uiTheme;
+    const { palette, buttonDisabled, buttonRaisedDisabled } = context.uiTheme;
```

Comment:
I think destructuring should be as follows:
```js
const {
  button,
  buttonFlat,
  buttonRaised,
  palette,
  buttonDisabled,
  buttonRaisedDisabled,
} = context.uiTheme;
```
You could also group a couple variables on one line, but destructuring twice is not really a pattern.